talos.tf

```diff
@@ -35,28 +35,30 @@ locals {
 
 data "talos_machine_configuration" "control_plane" {
   // enable although we have no control planes, to be able to debug the output
-  for_each         = { for control_plane in local.control_planes : control_plane.name => control_plane }
-  talos_version    = var.talos_version
-  cluster_name     = var.cluster_name
-  cluster_endpoint = local.cluster_endpoint
-  machine_type     = "controlplane"
-  machine_secrets  = talos_machine_secrets.this.machine_secrets
-  config_patches   = [yamlencode(local.controlplane_yaml[each.value.name])]
-  docs             = false
-  examples         = false
+  for_each           = { for control_plane in local.control_planes : control_plane.name => control_plane }
+  talos_version      = var.talos_version
+  cluster_name       = var.cluster_name
+  cluster_endpoint   = local.cluster_endpoint
+  kubernetes_version = var.kubernetes_version != null ? var.kubernetes_version : null
```

Comment:
I think this zero check is useless, don't you?

Reply:
Full agree!! 😄 .. just pushed an update